x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -595,7 +596,7 @@ void openSession(
                 threadPool,
                 responseListener,
                 ccrSettings.getRecoveryActionTimeout(),
-                ThreadPool.Names.GENERIC,
+                chunkResponseExecutor,
```

Comment:
Just double checking: in current code `chunkResponseExecutor` is `threadPool.generic()`, so nothing changes. But do we want the executor we pass to `RestoreSession` to use for timeouts to be always the same as `chunkResponseExecutor` (so this code is fine) or do we want to be always `threadPool.generic()`? I'm thinking, what happens if in the future we change `chunkResponseExecutor` to be something else? Are we still good?

Reply:
That's a good question. My intention here was to handle the timeout on the same threadpool as the non-timeout response. But you're right, the response is not a _chunk_ response so we shouldn't be using the chunk response executor here. In fact we handle the non-timeout response on the CCR pool, so I added a comment to suggest we come back and move this to the CCR pool too.